app/routes/goals.py

```diff
@@ -0,0 +1,124 @@
+from flask import Blueprint, request, jsonify, abort, make_response
+from app import db
+from app.models.goal import Goal
+from app.models.task import Task
+from sqlalchemy import desc, asc
+from app.routes.helper_function import get_goal_or_abort, validate_title_key_for_post_or_update, get_task_or_abort
+
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+# Wave 5
+@goals_bp.route("", methods=["GET"])
+def read_all_goal():
+    """
+        - Getting all goals and returning in json response with 200
+        - Returning empty list if no goal in database
+    """
+    params = request.args
+    # sort by title
+    if "sort" in params:
+        if params["sort"].lower() == "desc" or params["sort"].lower() == "descending":
+                chosen_goal = Goal.query.order_by( desc(Goal.title) ).all()
+        else:
+            chosen_goal = Goal.query.order_by( asc(Goal.title) ).all()
```

Comment:
Style: This should be `chosen_goals` since it's a collection we give it a plural name:

```suggestion
            chosen_goals = Goal.query.order_by( asc(Goal.title) ).all()
```